### PR TITLE
ARROW-13544 [Java]: Remove APIs that have been deprecated for long (Changes to JDBC)

### PR DESCRIPTION
**Pull request by liyafan82**
See https://issues.apache.org/jira/browse/ARROW-13544

According to the discussion in #10864 (comment), we want to split the task into multiple parts.

This PR is for the changes related to the JDBC adapter

### CODE REVIEW

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/AbstractJdbcToArrowTest.java [hunk 1]
```diff
@@ -140,4 +146,160 @@ public void destroy() throws SQLException {
...
+   * @param connection Database connection to be used. This method will not close the passed connection object. Since
+   *                   the caller has passed the connection object it's the responsibility of the caller to close or
+   *                   return the connection to the pool.
+   * @param query      The DB Query to fetch the data.
+   * @param allocator  Memory allocator
+   * @return Arrow Data Objects {@link VectorSchemaRoot}
+   * @throws SQLException Propagate any SQL Exceptions to the caller after closing any resources opened such as
+   *                      ResultSet and Statement objects.
+   */
+  public static VectorSchemaRoot sqlToArrow(Connection connection, String query, BufferAllocator allocator)
```
Comment (emkornfield) [inline java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/AbstractJdbcToArrowTest.java:164]: It seems ashame to simply do a code move here, would it make sense to make the deprecated methods package private instead and commnent that they are test only?

I think this also reminds me that the model for the iteration here doesn't match the Loader/Unloader pattern but creates a new VectorSchemaRoot each time (I might be misremembering though)

**Comment by emkornfield** [comment]
CC @xhochy I think last time we talked about this you were able to use the undeprecated methods, just  wanted to make sure this is still the case or you have a migration plan.

File: java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/AbstractJdbcToArrowTest.java (hunk 1 above)
Comment (liyafan82) [reply java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/AbstractJdbcToArrowTest.java:166]: > It seems ashame to simply do a code move here, would it make sense to make the deprecated methods package private instead and commnent that they are test only?
> 
I think it is a good idea to explicitly state that these methods are for test only. However, it is not feasible to declare them as package private, as they are used in at least two packages: `org.apache.arrow.adapter.jdbc` and `org.apache.arrow.adapter.jdbc.h2`.

> I think this also reminds me that the model for the iteration here doesn't match the Loader/Unloader pattern but creates a new VectorSchemaRoot each time (I might be misremembering though)

 I think you are right

**Comment by emkornfield** [comment]
+1 thank you.

**Pull request closed without merging**